src/libraries/Common/src/Interop/Windows/Kernel32/Interop.GetVolumeInformationByHandle.cs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class Kernel32
+    {
+        [LibraryImport(Libraries.Kernel32, EntryPoint = "GetVolumeInformationByHandleW", SetLastError = true, StringMarshalling = StringMarshalling.Utf16)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static unsafe partial bool GetVolumeInformationByHandleW(
+            SafeFileHandle hFile,
+            char* volumeName,
+            int volumeNameBufLen,
+            int* volSerialNumber,
+            int* maxFileNameLen,
+            int* fileSystemFlags,
+            char* fileSystemName,
+            int fileSystemNameBufLen);
+
+        public static unsafe int GetVolumeInformationByHandle(
+            SafeFileHandle hFile,
+            out string? volumePath,
+            bool wantsVolumePath,
+            int* volSerialNumber,
+            int* maxFileNameLen,
+            out int fileSystemFlags,
+            char* fileSystemName,
+            int fileSystemNameBufLen)
+        {
+            // Allocate output buffer on the stack initially.
+            const int stackAllocation = 512;
+            Span<char> volumePathBuffer = wantsVolumePath ? stackalloc char[stackAllocation + 1] : stackalloc char[0]; // +1 to ensure a \0 at the end, todo: is this necessary
```

Comment:
I don't think you need to implement the "wantsVolumePath" path until you have a caller that will pass in true. It's just dead code.

Reply:
Yes, there's lots of dead code in this currently (I have tried to not commit most of it lol). It was used in some older iteration locally.